eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -15,8 +18,9 @@ steps:
       version: "2.1.x"
   - task: Cache@2
     inputs:
-      key: 'nuget | "$(Agent.OS)" | $(Build.SourcesDirectory)/eng/Packages.Data.props'
+      key: 'nuget | "$(Agent.OS)" | $(Build.SourcesDirectory)/eng/Packages.Data.props | ${{parameters.NuGetCacheKey}}'
```

Comment:
You could use `$(Agent.JobName)` as well to keep the cache unique per job. We do that in Java https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/steps/cache-maven-repository.yml#L12.

Reply:
I think sharing it between tests on the same OS is fine.